doc/mgmt/SONiC_IntfRangeFeature_HLD.md

```diff
@@ -0,0 +1,235 @@
+Example: `ETHER_INTERFACE_RANGE` for physical interfaces<br>
+```
+<PARAM
+    name="phy-if-name"
+    help="Physical interface range"
+    ptype="ETHER_INTERFACE_RANGE"
+/>
+```
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+### 3.2.2 APP DB
+### 3.2.3 STATE DB
+### 3.2.4 ASIC DB
+### 3.2.5 COUNTER DB
+
+## 3.3 Switch State Service Design
+### 3.3.1 Orchestration Agent
+### 3.3.2 Other Process
+## 3.4 SyncD
+## 3.5 SAI
+## 3.6 User Interface
+### 3.6.1 Data Models
+NA
+### 3.6.2 CLI
+#### 3.6.2.1 Interface range commands in native mode
+
+####  3.6.2.1.1 Ethernet interface range:
+##### Configure physical interfaces in given range
+Syntax: `interface range Ethernet port - port[,port - port...]`
+```
+sonic(config)# interface range ?
+Ethernet         Physical interface range
+Vlan             Vlan interface range
+PortChannel      PortChannel interface range
+create           Create interfaces in range
+
+sonic(config)# interface range Ethernet 1-4,5
+sonic(conf-if-range-Eth1-4,5)# ?
+channel-group   Configure PortChannel parameters
+end             Exit to the exec Mode
+exit            Exit from current mode
+link            Interface link                   
+mac             Configure MAC parameters
+mtu             Configure MTU        
+nat-zone        NAT Zone               
+no              Negate a command or set its defaults
+service-policy  Apply ingress or egress policy      
+sflow           Interface sFlow settings
+shutdown        Disable the interface     
+spanning-tree   Configure spanning tree on interface      
+storm-control   configure storm-control   
+switchport      Configure switchport parameters      
+threshold       Configure priority-group/queue threshold on an interface.
+udld            UDLD interface configuration subcommands
+
+sonic(conf-if-range-Eth1-4,5)# mtu 9000
+sonic(conf-if-range-Eth1-4,5)#
+```
+####  3.6.2.1.2 Vlan interface range:
+###### Configure Vlan interfaces existing in given range
+Syntax: `interface range Vlan Vlan-id - Vlan-id[,Vlan-id - Vlan-id...]`
+```
+sonic(config)# interface range Vlan 1-20
+sonic(conf-if-range-Vl-1-20)# ?
+end                Exit to the exec Mode
+exit               Exit from current mode
+link               Interface link             
+mac                Configure MAC parameters   
+mtu                Configure MTU
+nat-zone           NAT Zone   
+neigh_suppress     Enable ARP and ND Suppression      
+no                 Negate a command or set its defaults
+service-policy     Apply ingress or egress policy
+
+sonic(conf-if-range-Vl-1-20)# mtu 9000
+sonic(conf-if-range-Vl-1-20)#
+```
+###### Create Vlan interfaces in given range
+Syntax: `interface range create Vlan Vlan-id - Vlan-id[,Vlan-id - Vlan-id...]`
+```
+sonic(config)# interface range create ?
+Vlan             Vlan interface range
+PortChannel      PortChannel interface range
+
+sonic(config)# interface range create Vlan 3-4,10,11
+sonic(conf-if-range-Vl-3-4,10,11)#
```

Comment:
A dynamic prompt for interface range could end up quite long string in case of non-contiguous range.
It would be even cumbersome to CLI scripts, including Skytest, to manipulate the expected prompt.

Can we make a simple prompt like Cisco?
e.g.
```
Device(config-if)# interface range fastethernet 5/1 - 5 , gigabitethernet 1/1 - 2 
Device(**config-if-range**)# no shutdown 
```

Reply:
agreed, have changed the CLI.